dev/com.ibm.ws.jaxrs.2.0_fat/fat/src/com/ibm/ws/jaxrs20/fat/RestMetricsTest.java

```diff
@@ -184,7 +189,20 @@ public void testCoreMethods() throws IOException {
         }
     }
 
-    //    @AllowedFFDC("com.ibm.ws.jaxrs.fat.restmetrics.MetricsUnmappedCheckedException")
+    @Test
+    public void testAbort() throws IOException {
+
+        runAbortGetMethod(ABORT_METHOD_INDEX, 200, "/restmetrics/rest/restmetrics/abortTest", "Metrics!");
+
+        // Expect a 200 since metric from other methods have already been collected.
```

Comment:
This sounds risky... JUnit doesn't offer any guarantees on which methods will be executed in which order, so it's possible that this one might be executed first.  If you want to avoid intermittent build breaks, I would suggest adding some code at the beginning of this method (on the the `@Before` method) to ensure that there are some metrics that have been collected, so that you can always count on them being there.

Reply:
As a test, what happens if you run this test in isolation - `./gradlew com.ibm.ws.jaxrs.2.0_fat:buildandrun -Dfat.test.class.name=RestMetricsTest -Dfat.test.method.name=testAbort`?

Reply:
Good catch!  I messed around with @Before but since these tests can run asynchronously even that would have intermittent failures without adding some synch blocks.  I didn't want to slow things down any more than I had to so instead I added a change to allow the 404 to be tolerated only when testAbort is running.